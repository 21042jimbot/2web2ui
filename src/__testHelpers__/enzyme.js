export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
