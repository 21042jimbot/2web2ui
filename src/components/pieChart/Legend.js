import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@sparkpost/matchbox';
import LegendItem from './LegendItem';

import styles from './Legend.module.scss';

const Legend = ({
  headerData,
  primaryData,
  secondaryData,
  onMouseOver,
  onMouseOut,
  onClick,
  hoveredItem
}) => {
  const hoveredDataSet = hoveredItem && hoveredItem.dataSet;

  const primaryLegend = primaryData.map((item, i) => (
    <LegendItem key={i}
      onMouseOver={() => onMouseOver(item, 'primary')}
      onMouseOut={onMouseOut}
      onClick={() => onClick(item)}
      hovered={hoveredDataSet === 'primary' && hoveredItem.index === i}
      otherHovered={hoveredItem && ((hoveredDataSet === 'primary' && hoveredItem.index !== i) || hoveredDataSet !== 'primary') }
      {...item} />
  ));

  const secondaryLegend = secondaryData
    ? <Grid.Column>
      {secondaryData.map((item, i) => (
        <LegendItem key={i}
          onMouseOver={() => onMouseOver(item, 'secondary')}
          onMouseOut={onMouseOut}
          onClick={() => onClick(item)}
          hovered={hoveredDataSet === 'secondary' && hoveredItem.index === i}
          otherHovered={hoveredItem && ((hoveredDataSet === 'secondary' && hoveredItem.index !== i) || hoveredDataSet !== 'secondary') }
          {...item} />
      ))}
    </Grid.Column>
    : null;

  const header = headerData.map((item, i) => <LegendItem key={i} {...item} />);

  return (
    <div>
      <div className={styles.LegendHeader}>{header}</div>
      <Grid>
        <Grid.Column>{primaryLegend}</Grid.Column>
        {secondaryLegend}
      </Grid>
    </div>
  );
};

Legend.propTypes = {
  headerData: PropTypes.array,
  primaryData: PropTypes.array,
  secondaryData: PropTypes.array,
  onMouseOver: PropTypes.func,
  onMouseOut: PropTypes.func,
  onClick: PropTypes.func,
  hoveredItem: PropTypes.object
};

Legend.displayName = 'PieChart.Legend';
export default Legend;
