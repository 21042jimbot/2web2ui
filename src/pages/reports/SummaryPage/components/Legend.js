import React, { Component } from 'react';

import styles from './Legend.module.scss';

class Legend extends Component {
  renderMetric(metric) {
    return (
      <div className={styles.Metric} key={metric.name}>
        <span className={styles.Color} style={{ backgroundColor: metric.stroke }}/>
        { metric.label }
      </div>
    );
  }

  render() {
    const { metrics } = this.props;

    return (
      <div className={styles.Legend}>
        { metrics && metrics.map((metric) => this.renderMetric(metric)) }
      </div>
    );
  }
}

export default Legend;
