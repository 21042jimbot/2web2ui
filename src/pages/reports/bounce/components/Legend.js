import React from 'react';
import { Grid } from '@sparkpost/matchbox';
import LegendItem from './LegendItem';

import styles from './Legend.module.scss';

const Legend = ({
  headerData,
  primaryData,
  secondaryData,
  handleMouseOver,
  handleMouseOut,
  handleClick
}) => {
  const primaryLegend = primaryData.map((item, i) => (
    <LegendItem key={i}
      onMouseOver={() => handleMouseOver(item, 'primary')}
      onMouseOut={handleMouseOut}
      onClick={() => handleClick(item)}
      {...item} />
  ));

  const secondaryLegend = secondaryData
    ? <Grid.Column>
        {secondaryData.map((item, i) => (
          <LegendItem key={i}
            onMouseOver={() => handleMouseOver(item, 'secondary')}
            onMouseOut={handleMouseOut}
            onClick={() => handleClick(item)}
            {...item} />
        ))}
      </Grid.Column>
    : null;

  const header = headerData.map((item, i) => <LegendItem key={i} {...item} />);

  return (
    <div>
      <div className={styles.LegendHeader}>{header}</div>
      <Grid>
        <Grid.Column>{primaryLegend}</Grid.Column>
        {secondaryLegend}
      </Grid>
    </div>
  );
};

export default Legend;
