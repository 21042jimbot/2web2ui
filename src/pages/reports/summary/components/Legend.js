import React from 'react';
import styles from './Legend.module.scss';

function renderMetric(metric) {
  return (
    <div className={styles.Metric} key={metric.name}>
      <span className={styles.Color} style={{ backgroundColor: metric.stroke }}/>
      { metric.label }
    </div>
  );
}
export default function Legend(props) {
  const { metrics } = props;

  return (
    <div className={styles.Legend}>
      { metrics && metrics.map(renderMetric) }
    </div>
  );
}
