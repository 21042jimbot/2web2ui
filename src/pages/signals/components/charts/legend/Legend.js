import React from 'react';
import { Tooltip } from '@sparkpost/matchbox';
import styles from './Legend.module.scss';

const Item = ({ label, fill = 'whitesmoke', tooltipContent }) => {
  const content = (
    <div className={styles.Item}>
      <span className={styles.Fill} style={{ background: fill }}/>
      <span className={styles.Label}>{label}</span>
    </div>
  );

  if (tooltipContent) {
    return (
      <Tooltip content={tooltipContent(label)} dark>
        {content}
      </Tooltip>
    );
  }

  return content;
};

const Legend = ({ items, tooltipContent }) => (
  <div>
    {items.map((item, i) => <Item key={i} tooltipContent={tooltipContent} {...item} />)}
  </div>
);

export default Legend;
