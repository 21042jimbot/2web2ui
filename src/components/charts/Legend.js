import React from 'react';
import styles from './Legend.module.scss';
import { Tooltip } from 'src/components/matchbox';
import cx from 'classnames';

const Item = ({ label, fill = 'whitesmoke', tooltipContent, hasBorder }) => {
  const content = (
    <div className={styles.Item}>
      <span className={cx(styles.Fill, hasBorder && styles.Border)} style={{ background: fill }} />
      <span className={styles.Label}>{label}</span>
    </div>
  );

  if (tooltipContent) {
    return (
      <Tooltip content={tooltipContent(label)} dark>
        {content}
      </Tooltip>
    );
  }

  return content;
};

const Legend = ({ items, tooltipContent }) => (
  <div className={styles.Legend}>
    {items.map((item, i) => (
      <Item key={i} tooltipContent={tooltipContent} {...item} />
    ))}
  </div>
);

export default Legend;
