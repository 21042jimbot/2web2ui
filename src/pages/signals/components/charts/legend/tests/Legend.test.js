import { shallow } from 'enzyme';
import React from 'react';
import Legend from '../Legend';

describe('Legend Component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      items: [
        { fill: 'blue', label: 'label 1' },
        { label: 'label 2' }
      ]
    };
    wrapper = shallow(<Legend {...props}/>);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Item').forEach((item) => {
      expect(item.dive()).toMatchSnapshot();
    });
  });

  it('renders correctly with tooltip content', () => {
    wrapper.setProps({ tooltipContent: (label) => label });
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Item').forEach((item) => {
      expect(item.dive()).toMatchSnapshot();
    });
  });
});
