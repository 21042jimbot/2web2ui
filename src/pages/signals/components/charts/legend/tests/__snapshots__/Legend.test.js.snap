// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend Component renders correctly 1`] = `
<div>
  <Item
    fill="blue"
    key="0"
    label="label 1"
  />
  <Item
    key="1"
    label="label 2"
  />
</div>
`;

exports[`Legend Component renders correctly 2`] = `
<div
  className="Item"
>
  <span
    className="Fill"
    style={
      Object {
        "background": "blue",
      }
    }
  />
  <span
    className="Label"
  >
    label 1
  </span>
</div>
`;

exports[`Legend Component renders correctly 3`] = `
<div
  className="Item"
>
  <span
    className="Fill"
    style={
      Object {
        "background": "whitesmoke",
      }
    }
  />
  <span
    className="Label"
  >
    label 2
  </span>
</div>
`;

exports[`Legend Component renders correctly with tooltip content 1`] = `
<div>
  <Item
    fill="blue"
    key="0"
    label="label 1"
    tooltipContent={[Function]}
  />
  <Item
    key="1"
    label="label 2"
    tooltipContent={[Function]}
  />
</div>
`;

exports[`Legend Component renders correctly with tooltip content 2`] = `
<Tooltip
  content="label 1"
  dark={true}
>
  <div
    className="Item"
  >
    <span
      className="Fill"
      style={
        Object {
          "background": "blue",
        }
      }
    />
    <span
      className="Label"
    >
      label 1
    </span>
  </div>
</Tooltip>
`;

exports[`Legend Component renders correctly with tooltip content 3`] = `
<Tooltip
  content="label 2"
  dark={true}
>
  <div
    className="Item"
  >
    <span
      className="Fill"
      style={
        Object {
          "background": "whitesmoke",
        }
      }
    />
    <span
      className="Label"
    >
      label 2
    </span>
  </div>
</Tooltip>
`;
